cli/command/container/hijack.go

```diff
@@ -29,6 +34,15 @@ func holdHijackedConnection(ctx context.Context, streams command.Streams, tty bo
 				restoreTerminal(streams, inputStream)
 			})
 		}()
+
+		// Wrap the input to detect detach control sequence.
+		// Use default detach sequence if an invalid sequence is given.
+		escapeKeys, err := term.ToBytes(detachKeys)
+		if len(escapeKeys) == 0 || err != nil {
+			escapeKeys = defaultEscapeKeys
+		}
+
+		inputStream = ioutils.NewReadCloserWrapper(term.NewEscapeProxy(inputStream, escapeKeys), inputStream.Close)
```

Comment:
This function is already over the cyclomatic complexity limit, so it would be great if we could avoid adding to the complexity.

I think this entire `inputStream` block could be a separate function:

```go

func setupInputStream(streams command.Streams, detachKeys string, tty bool) (inputStream io.ReadCloser, restore func(), err error) {
    if stream.In() == nil || !tty {
        return stream.In(), func(), nil
    }
    ...

    restore := func() {
        restoreOnce.Do(func() {
            restoreTerminal(streams, inputStream)
        })
    }
    return inputStream, restore, nil
}
```

This `restore` function can be used to replace the two duplicate `restoreTerminal(streams, inputStream)` calls in this function, along with their condition blocks, because we've already checked the condition once.

This new function should reduce the complexity by at least 5.